sys/cheri/cheri_sysctl.c

```diff
@@ -80,6 +80,26 @@ bool security_cheri_lib_based_c18n_default = false;
 SYSCTL_BOOL(_security_cheri, OID_AUTO, lib_based_c18n_default, CTLFLAG_RWTUN,
     &security_cheri_lib_based_c18n_default, 0,
     "Userspace library-based compartmentalisation default");
+
+/*
+ * Wrap function pointers when userspace library-based compartmentalisation
+ * (c18n) is enabled.
+ */
+bool security_cheri_lib_based_c18n_wrap_fptr = false;
+SYSCTL_BOOL(_security_cheri, OID_AUTO, lib_based_c18n_wrap_fptr, CTLFLAG_RWTUN,
+    &security_cheri_lib_based_c18n_wrap_fptr, 0,
+    "Wrap function pointers when userspace library-based compartmentalisation "
+    "is enabled");
+
+/*
+ * Forbid system calls from non-libsys code when userspace library-based
+ * compartmentalisation (c18n) is enabled.
+ */
+bool security_cheri_lib_based_c18n_secure_syscall = false;
```

Comment:
Is the issue that we're removing the permission or that we're enforcing its presence? I was assuming you'd disable enforcement, but could be persuaded this is the right way around.

Reply:
Yes, disabling enforcement is the better approach.